### PR TITLE
Fix broken links in docs

### LINKED ISSUE #1634: Broken links to some pages in docs
I've created this issue to track the progress of fixing broken links on some pages. 

I can successfully reproduce this weird behaviour:

1. Go to https://premake.github.io/docs/Developing-Modules/
2. Links are broken, because they are referencing `/docs/Developing-Modules/*` instead of `/docs/*`
3. Use sidebar navigation and go to **Adding a New Action**
4. Use sidebar navigation and go back to **Developing Modules**
5. Links are working fine
6. Refresh the page (F5)
7. Links are broken again

I will try do debug this but at the moment I suspect that it's Docusaurus' fault.

### PR DESCRIPTION
**Pull request by KyrietS**
**What does this PR do?**

Fix broken links in docs (#1634)

**How does this PR change Premake's behavior?**

*no*

**Anything else we should know?**

* I've added some additional information to website/README about referencing other pages in docs.

> ### Adding a reference to another documentation page
>
> Always reference another documentation page like this:
> ```markdown
> [some text](Case-Sensitive-Filename.md)
> ```
>
>and **never** like this:
> ```markdown
> [some text](some-markdown-file)
> [some text](/docs/some-markdown-file)
> [some text](https://premake.github.io/docs/some-markdown-file)
> ```

* As always I'm hosting a working demo to prove that this PR fixes the issue: 
https://kyriets.github.io/premake-core/docs/Developing-Modules

closes #1634

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [ ] Add unit tests showing fix or feature works; all tests pass
- [x] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits
- [x] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

File: website/docs/Building-Premake.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ If you downloaded a prebuilt binary package you can skip this page, which discus
 
 ## Using a Source Code Package ##
 
-If you have one of the [official source code packages](http://premake.github.io/download.html), you'll find that project files for a variety of toolsets have already been generated for you in the `build/` folder. Find the right set for your platform and toolset and build as you normally would  (i.e. run `make`). The resulting binaries will be placed in the top-level `bin/` folder.
+If you have one of the [official source code packages](/download), you'll find that project files for a variety of toolsets have already been generated for you in the `build/` folder. Find the right set for your platform and toolset and build as you normally would  (i.e. run `make`). The resulting binaries will be placed in the top-level `bin/` folder.
```
Comment (samsinsane) [inline website/docs/Building-Premake.md:10]: Just to confirm, this change is correct? (I can see that it works in your demo)

File: website/docs/Debugging-Scripts.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ title: Debugging Scripts
 Since Premake's update to 5.3, the only debugger that seems to be able to debug Premake is the free ZeroBrane Studio IDE.
```
Comment (samsinsane) [inline website/docs/Debugging-Scripts.md:7]: Unrelated, but we should update wording like this. It reads more like a community review, than a statement from the development team. @starkos

**Review by samsinsane — APPROVED** [review]

File: website/docs/Building-Premake.md (hunk 1 above)
Comment (KyrietS) [reply website/docs/Building-Premake.md:10]: Yes. This link `http://premake.github.io/download.html` is old and incorrect. Download page is stored in `website/src/pages/download.js` so it's completely valid to reference it using `/download`. I fixed it on the fly when I was reviewing all usages of markdown links in our docs.

Comment (samsinsane) [reply website/docs/Building-Premake.md:10]: Thanks for this explanation!

**Review by starkos — APPROVED** [review]
Thanks for taking care of this!

**Pull request merged**